CMakeLists.txt

```diff
@@ -7,10 +7,51 @@ if ( (${PROJECT_SOURCE_DIR} STREQUAL ${PROJECT_BINARY_DIR}) AND NOT ${INSOURCEBU
 endif ()
 
 if ( MSVC )
-  add_compile_options( /W4 )
-else()
-  add_compile_options( -Wall -Wextra -Wpedantic -Wno-missing-field-initializers )
-endif()
+    add_compile_options( /W4 )
+elseif ( "${CMAKE_CXX_COMPILER_ID}" MATCHES "Clang" )
+    add_compile_options (
+        -Wall
+        -Wextra
+        -pedantic
+        -Wshadow
+        -Wnon-virtual-dtor
+        -Wcast-align
+        -Wunused
+        -Woverloaded-virtual
+        -Wpedantic
+        -Wconversion
+        -Wsign-conversion
+        -Wmisleading-indentation
+        -Wnull-dereference
+        -Wdouble-promotion
+        -Wformat=2
+        -Wimplicit-fallthrough
+        -Wno-missing-field-initializers
+    )
+else ( "${CMAKE_CXX_COMPILER_ID}" MATCHES "GNU" )
+    add_compile_options (
+        -Wall
+        -Wextra
+        -pedantic
+        -Wshadow
+        -Wnon-virtual-dtor
+        -Wcast-align
+        -Wunused
+        -Woverloaded-virtual
+        -Wpedantic
+        -Wconversion
+        -Wsign-conversion
+        -Wmisleading-indentation
+        -Wduplicated-cond
+        -Wduplicated-branches
+        -Wlogical-op
+        -Wnull-dereference
+        -Wdouble-promotion
+        -Wformat=2
+        -Wimplicit-fallthrough
+        -Wno-missing-field-initializers
+    )
+endif ()
```

Comment:
Add **else** branch with an error of unsupported compiler (MSVC, Clang and GNU are supported).

Reply:
Well, I guess icc could compile this no problem, but ok